src/parquet_impl.cpp

```diff
@@ -948,9 +951,17 @@ OidFunctionCall1NullableArg(Oid functionId, Datum arg, bool argisnull)
 static List *
 get_filenames_from_userfunc(const char *funcname, const char *funcarg)
 {
+//pg16 update
+#if PG_VERSION_NUM > 160000
+    Node* minimalContext = new Node();
+    List* f = stringToQualifiedNameList(funcname, minimalContext);
+    delete minimalContext;
+#else
+    List* f = stringToQualifiedNameList(funcname);
+#endif
+
```

Comment:
It is necessary to pass `NULL` instead of a new context here:
```suggestion
#if PG_VERSION_NUM >= 160000
    List       *f = stringToQualifiedNameList(funcname, NULL);
#else
    List       *f = stringToQualifiedNameList(funcname);
#endif
```
It is necessary to pass `NULL` because otherwise `stringToQualifiedNameList()` will not throw an error and silently return `NIL` instead. See the comment:
https://github.com/postgres/postgres/blob/REL_16_3/src/include/utils/elog.h#L251-L260

Additionally the condition should be `PG_VERSION_NUM >= 160000` and it isn't necessary to add comment since the directive is self-explanatory.